application/org.openjdk.jmc.rjmx/src/main/java/org/openjdk/jmc/rjmx/triggers/internal/NotificationTrigger.java

```diff
@@ -238,7 +238,7 @@ public void triggerOn(IConnectionHandle connectionHandle, TriggerRule rule, MRIV
 			if (stateStore.m_lastTriggerErrorTimestamp == null || (aspectEvent.getTimestamp()
 					- stateStore.m_lastTriggerErrorTimestamp >= TRIGGER_ERROR_HANDLING_LIMIT_TIME_MS)) {
 				stateStore.m_lastTriggerErrorTimestamp = aspectEvent.getTimestamp();
-				handleException(connectionHandle, rule, e);
+				handleException(connectionHandle, rule, e, "");
```

Comment:
Nit: I'm one of these weird people who prefer sending null to indicate that there is no trigger message rather than having a message that happened to be empty, and then adopting the output when encountering that to something that is a bit more clear and unambiguous, e.g. "no trigger message available". If that is not what you want to go for, then perhaps maybe just set as message here something that explains why there is not trigger message.

Reply:
> Nit: I'm one of these weird people who prefer sending null to indicate that there is no trigger message rather than having a message that happened to be empty, and then adapting the output when encountering that to something that is a bit more clear and unambiguous, e.g. "no trigger message available". If that is not what you want to go for, then perhaps maybe just set as message here something that explains why there is not trigger message.

Hi @thegreystone, Thanks for pointing this out. I have changed the empty string to null and introduced a null check in the called method.